core/silkworm/types/block_test.cpp

```diff
@@ -188,4 +189,36 @@ TEST_CASE("EIP-1559 Header RLP") {
     CHECK(decoded == h);
 }
 
+TEST_CASE("Hash header boundary computation") {
+    BlockHeader h;
+    h.difficulty = intx::from_string<intx::uint256>("0xf90260f90207a068a61c4a05db4913009de5666753258eb9306157680dc5da0d");
+
+    auto boundary{h.boundary()};
+    auto boundary_view{full_view(boundary.bytes)};
+    auto expected_boundary{intx::bswap(
+        intx::from_string<intx::uint256>("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff") /
+        h.difficulty
+    )};
+    Bytes expected_boundary_view(kHashLength, '\0');
+    std::memcpy(&expected_boundary_view[0], intx::as_bytes(expected_boundary), kHashLength);
+
+    CHECK(boundary_view.compare(expected_boundary_view) == 0);
+}
+
+TEST_CASE("Hash header boundary computation when we have difficulty with power of 2") {
+    BlockHeader h;
+    h.difficulty = intx::from_string<intx::uint256>("0x10000000000");
+
+    auto boundary{h.boundary()};
+    auto boundary_view{full_view(boundary.bytes)};
+    auto expected_boundary{intx::bswap(
+        (intx::from_string<intx::uint256>("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff") /
+        h.difficulty) + 1u
+    )};
+    Bytes expected_boundary_view(kHashLength, '\0');
+    std::memcpy(&expected_boundary_view[0], intx::as_bytes(expected_boundary), kHashLength);
+
+    CHECK(boundary_view.compare(expected_boundary_view) == 0);
+}
+
```

Comment:
I see several problems here (sorry for being pedantic)

1. Please don't duplicate test cases : 1 test case ensemble one logic unit of test. If you need to test two (or more) different behaviors of the same unit use `SECTION`. See [examples](https://github.com/catchorg/Catch2/blob/devel/docs/tutorial.md#test-cases-and-sections)
2. The writing of this test is not relieable: `expected_boundary` is computed exactly as `header.boundary()` does so there absolutely no chance expected boundary might result different from computed boudary. What we need instead is compare a well known boundary value (or a set of) with the outcome of `header.boundary()` and see if we've done the function correctly;
3. Variable naming : if you write `Bytes expected_boundary_view(kHashLength, '\0');` I should expect the initialized variable to be a `ByteView` not a `Byte`
4. For coverage we should also implement the test for difficulty <= 1u

Reply:
now all of the above they should good.  (do not worry about being pedantic, it is for the greater good)